packages/metadata/lib/MetadataValidationState.ts

```diff
@@ -0,0 +1,20 @@
+import type { IMetadataValidationState } from '@comunica/types';
+
+/**
+ * Reusable implementation for metadata validation states.
+ */
+export class MetadataValidationState implements IMetadataValidationState {
+  private readonly invalidateListeners: (() => void)[] = [];
+  public valid = true;
```

Comment:
> What is the purpose of the valid property if it is never read?

It's not being read atm indeed, but it might very well be in the future, so I would keep it public.

> Should an error be thrown if invalidate is called when valid is already false.

That might be good indeed!